### PR TITLE
fix(frontend): `now` fragment should be singleton

### PR DESCRIPTION
**Pull request by jon-chuang**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
`now` fragment was not singleton

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
Fixes: https://github.com/risingwavelabs/risingwave/issues/7543

Fixes adding e2e test for now/temporal filter: https://github.com/risingwavelabs/risingwave/issues/6790

### CODE REVIEW

**Review by soundOfDestiny — APPROVED** [review]
LGTM

File: src/stream/src/executor/now.rs [hunk 1]
```diff
@@ -99,24 +99,27 @@ impl<S: StateStore> NowExecutor<S> {
                 let time_millis = Epoch::from(barrier.epoch.curr).as_unix_millis();
                 let timestamp = Some(ScalarImpl::Int64((time_millis * 1000) as i64));
 
-                let data_chunk = DataChunk::from_rows(
-                    &if last_timestamp.is_some() {
```
Comment (soundOfDestiny) [inline src/stream/src/executor/now.rs:103]: what if `last_timestamp` == `None` ?

**Review by soundOfDestiny — CHANGES_REQUESTED** [review]
rest lgtm

File: src/stream/src/executor/now.rs (hunk 1 above)
Comment (jon-chuang) [reply src/stream/src/executor/now.rs:103]: I just split the code upon last_timestamp.is_some() branch, it's easier to read IMO.

File: e2e_test/streaming/temporal_filter.slt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
+
+statement ok
+create table t1 (v1 timestamp);
+
+# This statement should be correct for the next ~1000 years
```
Comment (fuyufjh) [inline e2e_test/streaming/temporal_filter.slt:7]: I love this idea! ☺️ Furthermore, can we add more test cases into this `slt` file? Only simple inserts are tested now. I think it's better to add tests for updates/deletes, or add some cases with other varients of predicate (e.g. `BETWEEN`)

Comment (jon-chuang) [reply e2e_test/streaming/temporal_filter.slt:7]: Indeed, we should do so.

**Pull request merged**